src/ir/typescript_types.py

```diff
@@ -216,6 +232,86 @@ def get_name(self):
         return 'undefined'
 
 
+class NumberLiteralType(TypeScriptBuiltin):
+    def __init__(self, literal, name="NumberLiteralType", primitive=False):
+        super().__init__(name, primitive)
+        self.literal = literal
+        self.supertypes.extend([ObjectType(), NumberType()])
+
+    def get_literal(self):
+        return self.literal
+
+    def is_assignable(self, other):
+        return ((isinstance(other, NumberLiteralType) and
+                  other.get_literal() == self.get_literal()) or
+                    isinstance(other, NumberType))
+
+    def get_name(self):
+        return self.name
+
+
+class StringLiteralType(TypeScriptBuiltin):
+    def __init__(self, literal, name="StringLiteralType", primitive=False):
+        super().__init__(name, primitive)
+        self.literal = literal
+        self.supertypes.extend([ObjectType(), StringType()])
+
+    def get_literal(self):
+        return '"' + self.literal + '"'
+
+    def is_assignable(self, other):
```

Comment:
Could you please add a comment about when a type is a assignable to a string literal type?

Reply:
`is_assignable` returns True when the current (`self`) type is assignable to the `other`.
Regardless, I am adding a comment to describe the method.

Reply:
yes, indeed.
But can you comment about the specific case of num literal types (e.g., what does the condition below check?).

Reply:
Will do, just noted the above for clarity.